slave/process-hadoop-hbase/bin/process-hadoop_hbase.sh

```diff
@@ -0,0 +1,123 @@
+#! /bin/bash
+
+#
+# Manage HBase user namespaces in Hadoop environment
+#
+# In addition unix users and groups need to be managed on the whole Hadoop cluser.
+#
+# Where to launch:
+# * on master node (or any node with HBase client and admin client keytab)
+# * launch only once
+#
+# Requirements:
+# * client admin Kerberos keytab (not required for unsecured cluster)
+# * HBase client installed and configured
+#
+# Physical delete:
+# * disabled by default (enable using FORCE_DELETE=1)
+# * removes all the tables in user namespace and the namespace itself
+#
+# Last actions are stored in /var/tmp/perun-hadoop-hbase-*.sh.
+#
+# Usernames can't contain '-' character, they will by skipped.
+#
+
+PROTOCOL_VERSION='3.0.0'
+
+function process() {
+	FROM_PERUN="${WORK_DIR}/hadoop_hbase"
+
+	FORCE_DELETE=${FORCE_DELETE:-'0'}
+	KEYTAB=${KEYTAB:-'/etc/security/keytab/hbase.service.keytab'}
+	PRINCIPAL=${PRINCIPAL:-"hbase/`hostname -f`@${REALM}"}
+	RESERVED=${RESERVED:-'NAMESPACE|default|hbase|hive|oozie'}
+
+	KRB5CCNAME="FILE:${WORK_DIR}/krb5cc_perun_hadoop_hbase"
+
+	I_HBASE_CREATED=(0 'User ${login} will be created.')
+	I_HBASE_DELETED=(0 'User ${login} will be deleted.')
+	I_HBASE_OK=(0 'HBase shell OK.')
+	I_HBASE_SIMULATED=(0 'Delete actions not performed.')
+	E_KINIT_FAILED=(1 'Kinit on HBase master failed')
+	E_HBASE_NAMESPACE_LIST=(2 'Cannot get list of namespaces')
+	E_HBASE_TABLE_LIST=(3 'Cannot get list of tables from ${login}')
+	E_HBASE_FAILED=(4 'HBase shell failed')
+	E_EMPTY_LIST=(5 'The list is empty!')
+	E_EMPTY_USERNAME=(6 'Empty username')
+
+	create_lock
+	chown hbase "${WORK_DIR}"
+
+	# Kerberos ticket (only for secured cluster)
+	if [ -n "$REALM" ]; then
+		su hbase -s /bin/bash -p -c "kinit -k -t '${KEYTAB}' '${PRINCIPAL}'" || log_msg E_KINIT_FAILED
+	fi
+
+	# get list from HBase
+	echo list_namespace | su hbase -s /bin/bash -p -c "hbase shell -n" >"${WORK_DIR}/out.txt" 2>/dev/null || log_msg E_HBASE_NAMESPACE_LIST
+	head -n -3 "${WORK_DIR}/out.txt" | sort > "${WORK_DIR}/hbase-list.txt"
+
+	# get list from Perun
+	[ -s "${FROM_PERUN}" ] || log_msg E_EMPTY_LIST
+	# HBase can't handle '-' in login names, blacklist all such users
+	grep -v -- '-' "${FROM_PERUN}" | sort > "${WORK_DIR}/perun-list.txt"
+
+	# compare and action
+	rm -f "${WORK_DIR}/add.hbase" "${WORK_DIR}/del.hbase"
```

Comment:
Work DIR is always extracted tar from Perun server so there will be no add.hbase and del.habse file at start of the script.